### PR TITLE
Add r4 instance types for AWS

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
### Description

Added the following AWS types:

- r4.large
- r4.xlarge
- r4.2xlarge
- r4.4xlarge
- r4.8xlarge
- r4.16xlarge

Also updated pricing information.

### Status

- done, ready for review

### Checklist (tick everything that applies)

- [ ] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

File: contrib/scrape-ec2-prices.py [hunk 1]
```diff
@@ -101,12 +110,15 @@
 REGION_NAME_MAP = {
     'us-east': 'ec2_us_east',
     'us-east-1': 'ec2_us_east',
+    'us-east-2': 'ec2_us_east',
```
Comment (amisstea) [inline contrib/scrape-ec2-prices.py]: I believe the value should match up to the `api_name` attribute of the region. Therefore, should this be `ec2_us_east_ohio` instead?

File: contrib/scrape-ec2-prices.py [hunk 2]
```diff
@@ -101,12 +110,15 @@
...
     'us-west': 'ec2_us_west',
     'us-west-1': 'ec2_us_west',
     'us-west-2': 'ec2_us_west_oregon',
     'eu-west-1': 'ec2_eu_west',
+    'eu-west-2': 'ec2_eu_west',
```
Comment (amisstea) [inline contrib/scrape-ec2-prices.py]: If I'm not mistaken, this region is yet to be defined in the driver. That being said, it's worth having but the value should probably be different from the one mapped to by `eu-west-1`. How `ec2_eu_west_2` instead?

**Comment by tonybaloney** [comment]
thanks for the contribution @jdeppe-pivotal there is a small merge conflict to resolve first. please rebase and push a merge commit

**Comment by tonybaloney** [comment]
the build error is on trunk. changes are good to merge 👍

**Pull request merged**